extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -74,6 +74,10 @@ class FlywayProcessor {
 
     private static final Logger LOGGER = Logger.getLogger(FlywayProcessor.class);
 
+    private static final Set<String> BANNED_PLUGINS = Set.of(
+            "org.flywaydb.core.internal.configuration.resolvers.GoogleSecretsResolver",
+            "org.flywaydb.core.internal.configuration.resolvers.HashicorpVaultResolver");
```

Comment:
Wondering if we should support these if the extensions supporting them are around? It could be done easily if they provide capabilities. But I'm not sure that would be enough to make them work.

/cc @vsevel @loicmathieu

Reply:
Google Secret Manager didn't provides capabilities yet. But if we suport Google Secret Manager inside Flyway but not as datasource credentials provider it would be inconsistent so first we will need to support it as credentials provider as Vault aready support. What surprised me is that in their documentation there is no Google Credentials support so it may only works when running inside a Google Cloud managed runtime, that restrict its usage a lot and may complicate local dev.

Reply:
I have not seen the source for `HashicorpVaultResolver`, but I guess they reimplemented a very limited http vault client (no auth, only the `get` to retrieve secrets).
see https://flywaydb.org/blog/integrating-vault-to-secure-flyway-parameters
I am not sure we want to go through supporting this.

I have not tested, but the first approach would be configuring vault with:
```
quarkus.vault.secret-config-kv-path=data/flyway/myconfig
```
and this vault secret content for instance:
```
quarkus.flyway.X=Y
```

this should make this property `X` available for the `flyway` extension (the `FlywayRuntimeConfig` is `phase = ConfigPhase.RUN_TIME`). vault is running in the bootstrap phase, so this should not conflict.

if it does, the 2nd approach is indeed to support consuming a credentials provider in the `flyway` extension (note that the vault credentials provider supports only one `kv-path` so it would not be able to read several paths as the `HashicorpVaultResolver` says it does).

given the flyway props exposed by quarkus, I do not see anything there that is really worth storing in vault.
since the datasource credentials are expressed through the `datasource` or `orm` extension, there is only may be the `flyway.licenseKey` that is worth it, but it is not part of the quarkus flyway configuration. so the first approach has limited interest, unless the user wants to use vault as a config manager (as opposed to a secret store).

in any ways, I do not see supporting the `HashicorpVaultResolver` (and its own vault http client - however it has been implemented) as the right approach.